src/TestFramework/Coverage/XmlReport/IndexXmlCoverageParser.php

```diff
@@ -111,14 +111,19 @@ private static function assertHasExecutedLines(SafeDOMXPath $xPath, bool $isForG
 
     private static function getProjectSource(SafeDOMXPath $xPath): string
     {
-        // PHPUnit >= 6
-        $sourceNodes = $xPath->query('//p:project/@source');
+        $sourceQueries = [
+            '//p:project/@source',  // PHPUnit >= 6
+            '//p:project/@name',    // PHPUnit < 6
+        ];
 
-        if ($sourceNodes->length > 0) {
-            return $sourceNodes[0]->nodeValue;
+        foreach ($sourceQueries as $sourceQuery) {
+            $source = $xPath->queryAttribute('//p:project/@source')?->nodeValue;
+
+            if ($source !== null) {
+                return $source;
+            }
         }
 
-        // PHPUnit < 6
-        return $xPath->query('//p:project/@name')[0]->nodeValue;
+        throw new RuntimeException('Could not find the source attribute for the project');
```

Comment:
I think it is a fix as previously it meant we ended up with a `TypeError` and now we have a `RuntimeException`. We may want to have a more specific and user friendly exception.

This should be extracted and covered by tests.